pyasic_api/get/__init__.py

```diff
@@ -83,3 +102,89 @@ async def get_ip_data(ip):
         "total_chips": data.total_chips,
         "ideal_chips": data.ideal_chips,
     }
+
+led_responses = {
+    200: {
+        "description": "Success",
+        "content": {
+            "application/json": {
+                "examples": {
+                    "on": {
+                        "summary": "Light Is On",
+                        "value": {"light_status": True, }
+                    },
+                    "off": {
+                        "summary": "Light is Off",
+                        "value": {"light_status": False, }
+                    },
+                }
+            }
+        }
+    },
+}
+
+@router.get("/{ip}/led/{led_mode}/",
+    summary="Check, toggle, or set the indicator LED state.",
+    response_description="The state of the LED following this operation.",
+    responses=led_responses)
+async def leds(ip, led_mode: LEDMode):
+    miner = await get_miner(ip)
+    light_status = await miner.check_light()
+    if led_mode is LEDMode.status: True
+    elif led_mode is LEDMode.toggle:
+        if light_status:
+            await miner.fault_light_off()
+            light_status = False
+        else:
+            await miner.fault_light_on()
+            light_status = True
+    elif led_mode is LEDMode.on:
+        await miner.fault_light_on()
+        light_status = True
+    elif led_mode is LEDMode.off:
+        await miner.fault_light_off()
+        light_status = False
+    else: True
+    return {"light_status": light_status,}
```

Comment:
How does that `if cond: True` work?  Syntactic sugar for a return?  It didn't seem to do anything when I tested it, so not sure if its just not backwards compatible with 3.10 or what.  In any case, would prefer to be explicit and backwards compatible.

In addition, there is no need to manually set `light_status`, you should just be able to return directly at that point.

```python
light_status = await miner.check_light()
if led_mode is LEDMode.status:
    return {"light_status": light_status}
elif led_mode is LEDMode.toggle:
    if light_status:
        if await miner.fault_light_off():
            return {"light_status": not light_status}
    else:
        if await miner.fault_light_on():
            return {"light_status": not light_status}
    # if the light fails to toggle, light status is the same.
    # could also raise HTTP error here
    return {"light_status": light_status}
elif led_mode is LEDMode.on:
    if await miner.fault_light_on():
        return {"light_status": True}
    raise Exception # Change this to HTTP error with an import and make sure it's the proper type
elif led_mode is LEDMode.off:
    if await miner.fault_light_off():
        return {"light_status": False}
    raise Exception # Change this to HTTP error with an import and make sure it's the proper type
```

Reply:
I was just reducing the number of final returns. I had it this way (less elegantly & minus the exceptions) originally. This does all work for me.

Reply:
I'm ok with whatever, I think its generally good to be more explicit in this case.  Its not really super important to me how we return the data, but the main concern I have is that we need to be very clear to any user that "THIS COMMAND FAILED", and I'm not sure whether returning a HTTP 4xx error is the best way to do that or specify some kind of error message.

I believe the 4xx errors are clearly defined in the docs, so my thought was to use the already defined errors instead of making our own, but fully open to whatever.

Reply:
Sounds good.